SPARCore/GADArticle.m

```diff
@@ -12,35 +11,74 @@ @implementation GADArticle
 static NSString *const API_PUBLICATION_ID = @"publication";
 static NSString *const API_TITLE = @"title";
 
-static NSString *const API_HOSTNAME = @"https://g2j7qs2xs7.execute-api.us-west-2.amazonaws.com/";
-static NSString *const API_PREFIX = @"devstable";
+static NSString *const API_ITMES = @"items";
+static NSString *const API_NEXT_PAGE_TOKEN = @"nextPageToken";
+
 static NSString *const API_PUBLICATION_PATH = @"publications";
 static NSString *const API_ARTICLE_PATH = @"articles";
 
-+ (void) articlesFromPublication: (NSString *)publicationId
-                                        completionHandler:(void(^_Nonnull)(NSArray<GADArticle *>
-                                                                           *_Nullable articles,
-                                                            NSError *_Nullable error))completion {
-    NSURL *queryURL = [GADArticle urlForArticlesFromPublication:publicationId];
++ (void) articlesForPublicationId: (NSString *)publicationId
+                   withCompletion:(void(^_Nonnull)(NSArray<GADArticle *>
+                                                   *_Nullable articles,
+                                                   NSError *_Nullable error))completion {
+    GADPublication *pub = [GADPublication new];
+    pub.publicationId=publicationId;
+    NSURL *queryURL = [pub urlForArticles];
     
-    [GADRemoteModel fetchModelsWithParams:queryURL
+    [super fetchModelsWithParams:queryURL
                           queryParameters:@{}
                          modelTransformer:^(NSData *jsonData) {
-                            return [GADArticle articlesFromJSON:jsonData];
+                            return [self articlesFromJSON:jsonData];
                          }
                         completionHandler:completion];
 
 }
 
++ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
+    
+    NSDictionary* jsonDict=[NSDictionary new];
+    NSArray *jsonArray=[NSArray new];
+    NSMutableArray <GADArticle *> *articles = [NSMutableArray new];
+    
+    NSError *JSONParsingError;
+    jsonDict = [NSJSONSerialization JSONObjectWithData:json
+                                                         options:kNilOptions error:&JSONParsingError];
+    
+    jsonArray=[jsonDict valueForKey:API_ITMES];
```

Comment:
what other keys are given to us here? Do we need to make a ticket to read and support them?